webapp/channels/src/components/admin_console/license_settings/trial_banner/trial_banner.tsx

```diff
@@ -94,7 +94,7 @@ const TrialBanner = ({
 
     const {formatMessage} = useIntl();
     const state = store.getState();
-    const getCategory = makeGetCategory();
+    const getCategory = useMemo(makeGetCategory, []);
```

Comment:
I wonder if it would be a good idea to use `useRef` for this sort of thing instead of `useMemo` for these cases. I doubt it has any real value on performance though.

Alternatively, I'm not completely sure what the sigature would look like, but I feel like making a `useSelectorFactory` would be good since we're doing this same pattern repeatedly. It probably needs some type parameters to make TS happy, but something like this might work
```js
function useSelectorFactory(factory, ...args) {
    const selector = useRef(factory());
    return useSelector((state) => selector.current(state, ...args));
}
```

Reply:
The reason why I wouldn't use `useRef` the way you have shown is that `factory` gets executed on every render, while the `useMemo` approach only executes it on the first render.

Reply:
Ah, I see. It's weird that `useRef` doesn't seem to support the same pattern that `useState` does where you can call it as `useState(() => initialValue())` to avoid calling the factory repeatedly, but I guess that's something with `useState` using a getter instead of returning an object